### PR TITLE
Bind server to RawFd

### LINKED ISSUE #105: Start server from RawFd
Cargo watch does not work because the sometimes the socket that's running will not be closed in time. That's why catflap exists. The problem is that the actix-web server does not allow for servers to be created from a raw file descriptor.

https://github.com/passcod/catflap  

There are integrations with hyper and iron but not with actix-web.

### PR DESCRIPTION
**Pull request by ghost**
Attempt to resolve #105

### CODE REVIEW

File: src/server/srv.rs [hunk 1]
```diff
@@ -173,9 +176,22 @@ impl<H> HttpServer<H> where H: IntoHttpHandler + 'static
         self.sockets.keys().cloned().collect()
     }
 
+    #[cfg(target_family="unix")]
+    /// The socket to bind to
+    ///
+    /// To bind multiple sockets this method can be called multiple times.
+    pub fn bind_socket(mut self, sock: RawFd) -> io::Result<Self> {
```
Comment (fafhrd91) [inline src/server/srv.rs:183]: Let’s rename it to a bind_raw_fd. I am not sure about safety though

**Comment by fafhrd91** [comment]
I don't think we need bind_raw_fd at all.
could you rewrite your example using HttpServer::listen() method

**Comment by fafhrd91** [comment]
Ping

**Comment by fafhrd91** [comment]
reopen if you would like to complete this pr

**Comment by ffimnsr** [comment]
this didn't get to release that's why I was searching on how to bind fd

**Comment by fafhrd91** [comment]
@ffimnsr Rawfd is not supported atm. Should be easy to support, but it should be added to actix-server first

**Pull request closed without merging**